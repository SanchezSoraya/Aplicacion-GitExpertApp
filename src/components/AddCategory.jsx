import { useState } from "react"


export const AddCategory = ({ setcategories }) => {
  //cada componente puede tener su propio estado
const [inputValue, setInputValue] = useState(''); 

const onInputChange = ( { target }) => {
  setInputValue( target.value );
}
const onSubmit = ( event ) =>{
  if( inputValue.trim().length <= 1) return;
  event.preventDefault();

  setcategories(categories => [inputValue, ...categories]);
  setInputValue(' ');
}

  return (
    <form onSubmit={onSubmit}>
      <input
        type="text"
        placeholder="Buscar gits"
        value={ inputValue }
        onChange={ onInputChange }
      />
    </form>
  )
}
