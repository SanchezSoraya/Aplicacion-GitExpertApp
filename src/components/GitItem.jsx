import  PropTypes  from "prop-types"



export const GitItem = ({title, url, id}) => {

  
  return (
    <div className="card">
      <img src={url} alt={title}/>
      <p>{title}</p>
    </div>
  )
}
GitItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
}

