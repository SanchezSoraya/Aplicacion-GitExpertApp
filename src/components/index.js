export {AddCategory} from './AddCategory';
export {GifGrid} from './GifGrid';
export {GitItem} from './GitItem';