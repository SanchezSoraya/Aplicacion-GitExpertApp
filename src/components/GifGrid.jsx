import GitItem from "./GitItem";
import { useFetchGifs } from "../hooks/useFetchGifs";

export const GifGrid = ({ category }) => {

  const { images, isLoading } = useFetchGifs( category );
  

  return (
    <>
      <h3>{category}</h3>
      <div className="card-grid">
        {
        images.map(( image ) => (
          <GitItem 
              key={image.id}
              {...image }
          />
        ))
        }
       
      </div>
      
    
    </>
  )
}
