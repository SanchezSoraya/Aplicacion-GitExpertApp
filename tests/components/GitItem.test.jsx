import { render, screen } from '@testing-library/react';
import { GitItem } from '../../src/components/GitItem';



describe('Pruebas en <GitItem /> ', () => { 
  const title = 'Saitama'; 
  const url = 'https://one-punch.com/saitama.jp';
test('debe de hacer mactch con el snapshot', () => {
  const { container } = render( <GitItem title={ title } url={ url } />);
  expect( container ).toMatchSnapshot();
  });

  test('debe de mostrar la imagen con el URL y el ALT indicado', () => {
    render(<GitItem title={ title } url={ url } />);  
    //screen.debug();
    //expect( screen.getByRole('img').src).toBe(url);
    //expect( screen.getByRole('img').alt).toBe(title);

    const {src, alt } = screen.getByRole('img'); 
    expect( src ).toBe( url ); 
    expect( alt ).toBe( alt ); 
  });

  test('debe de mostrar el titulo en el componente', () => {
    render( <GitItem title={ title } url={ url } />);
    expect( screen.getByText( title )).toBeTruthy();
  })

});