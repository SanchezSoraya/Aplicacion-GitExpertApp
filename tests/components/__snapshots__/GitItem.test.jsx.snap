// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GitItem />  debe de hacer mactch con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://one-punch.com/saitama.jp"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
